### PR TITLE
[ROBO-1053] Update for roboto 0.5; simplify

### PR DESCRIPTION
**Pull request by GabeMedrash**
**Context:**
Recent rounds of `roboto` SDK development resulted in new utilities for use with Roboto Actions. This updates our Python-based action cookiecutter to our latest code patterns, and also gets rid of some boilerplate that I'm betting was net distracting.

**Changeset:**
- Upgrade dev deps in repo.
- Change generated action to work with the latest and greatest Roboto SDK version; slim down action to its absolute barest.
- Updates to scripts used to run the action locally.
- Update base Docker image used by customer actions to Python `3.12-slim-bookworm`. This has implications for our "executor sidecar" containers--which ought to be updated to match--because they should be using the same base image to reduce the number of Docker layers that Fargate needs to download when running actions produced by our own cookiecutter.

**Testing:**
1. In a temp dir, in a fresh venv:
```
$ .venv/bin/python -m pip install cookiecutter
$ .venv/bin/cookiecutter git@github.com:roboto-ai/cookiecutter-roboto-actions.git --checkout feature/ROBO-1053-update-for-roboto-0.5 --directory="cookiecutter-roboto-python-action"
$ ./scripts/setup.sh && ./scripts/build.sh && ./scripts/run.sh && ROBOTO_ORG_ID=<...> ./scripts/deploy.sh
```
2. Once deployed to my dev stack, I verified that the action runs on our hosted compute as well.

**Reviewer questions:**
1. What do you think about the simplifications made to the action (the `__main__.py` file) produced by this cookiecutter?
2. Now that `dataset_id`, `invocation_id`, and `org_id` are required env vars, I had to add values for them to the `docker run` command in `./scripts/run.sh`. I set all to `"NOT_SET"`. What do you think? 
3. Also, because of how `ActionRuntime` works, a Roboto config file is now required (?), so I volume mounted `~/.roboto/config.json` in the `./scripts/run.sh` script. Seems fragile and weird to me, but, it works. I also don't know whether `./scripts/run.py` is genuinely useful.
4. Should we be pinning the version of `roboto` in the `requirements.txt` file produced by the action?

### CODE REVIEW

File: cookiecutter-roboto-python-action/{{cookiecutter.__project_slug}}/scripts/run.sh [hunk 1]
```diff
@@ -33,8 +27,14 @@ if [[ ! "$output_dir" = /* ]]; then
...
+    -u $(id -u):$(id -g) \
+    -v ~/.roboto/config.json:/roboto.config.json \
     -v $input_dir:/input \
     -v $output_dir:/output \
+    -e ROBOTO_CONFIG_FILE=/roboto.config.json \
     -e ROBOTO_INPUT_DIR=/input \
     -e ROBOTO_OUTPUT_DIR=/output \
+    -e ROBOTO_DATASET_ID=NOT_SET \
+    -e ROBOTO_INVOCATION_ID=NOT_SET \
+    -e ROBOTO_ORG_ID=NOT_SET \
```
Comment (jhubberts) [inline cookiecutter-roboto-python-action/{{cookiecutter.__project_slug}}/scripts/run.sh:39]: This is sooooo useful

**Review by jhubberts — APPROVED** [review]
This looks awesome!

**Comment by YvesSchoenberg** [comment]
Looking good! Tried it out locally and on the platform - works nicely. Concerning your questions:

1.) `__main__.py` is much simpler now - love it. 
2.) and 3.) I think that makes all sense for now. At some point, we may want to add a more first-class experience to run an Action locally, for example: `roboto actions invoke --local` - which is obviously tricky (especially for ingested-related Actions which are creating topics, adding tags etc.). The `run.sh` is a good first step.  
4.) I think having the latest SDK version here is fine. We just need to make sure to update the package, if we add incompatible changes to the SDK. 

By the way, I think adding a simple dummy integration test (similar to this one: https://github.com/roboto-ai/roboto-ingestion-actions/blob/main/actions/csv_ingestion/scripts/integration_test.py) could also be useful.

**Review by YvesSchoenberg — APPROVED** [review]

**Comment by GabeMedrash** [comment]
> By the way, I think adding a simple dummy integration test (similar to this one: https://github.com/roboto-ai/roboto-ingestion-actions/blob/main/actions/csv_ingestion/scripts/integration_test.py) could also be useful.

Yeah, I think that's a good idea. It seems like having two types of tests would be helpful:
1. We could put some sort of e2e test in this cookiecutter repo, which would:
    a. Run `cookiecutter` to create a test action.
    b. Build/deploy that test action into `beta` or something like that (and also create a dataset and upload a file to it)
    c. Run the action and assert that it completed successfully

This would tell us that this `cookiecutter` (and therefore, `roboto actions init`) works as expected, and continues to work over time. We could put it into a GH Action and run it once a day.

2. Put a test in the action generated by this cookiecutter. This is a bit more complicated to do in a meaningful way, considering that doing much of anything in the test would create test artifacts in customer accounts. We could also just set up a test harness for them, and have the test do nothing. Many other repo-starters that I've used in the past do this, though I tend to ignore those test harnesses.

I'll merge this as-is, and can capture the testing story in a JIRA.

**Pull request merged**